src/components/SearchableSelect/SearchableSelect.jsx

```diff
@@ -0,0 +1,427 @@
+
+	reducers,
+
+	components: {
+		/**
+		 * Content this is displayed when nothing is selected.
+		 */
+		Placeholder: createClass({
+			displayName: 'SearchableSelect.Placeholder',
+			propName: 'Placeholder',
+		}),
+		/**
+		 * A selectable option in the list.
+		 */
+		Option: createClass({
+			displayName: 'SearchableSelect.Option',
+			propName: 'Option',
+			propTypes: DropMenu.Option.propTypes,
+		}),
+		/**
+		 * Groups `Option`s together with a non-selectable heading.
+		 */
+		OptionGroup: createClass({
+			displayName: 'SearchableSelect.OptionGroup',
+			propName: 'OptionGroup',
+			propTypes: DropMenu.OptionGroup.propTypes,
+		}),
+	},
+
+	propTypes: {
+		/**
+		 * Should be instances of {`SearchableSelect.Placeholder`, `SearchableSelect.Option`, `SearchableSelect.OptionGroup`}. Other direct child elements will not render.
+		 */
+		children: node,
+		/**
+		 * Appended to the component-specific class names set on the root elements. Applies to *both* the control and the flyout menu.
+		 */
+		className: string,
+		/**
+		 * Styles that are passed through to root element.
+		 */
+		style: object,
+		/**
+		 * Allows user to reset the `optionIndex` to `null` if they select the placeholder at the top of the options list.
+		 * If `false`, it will not render the placeholder in the menu.
+		 */
+		hasReset: bool,
+		/**
+		 * Disables the SearchableSelect from being clicked or focused.
+		 */
+		isDisabled: bool,
+		/**
+		 * Displays a centered LoadingIcon to allow for asynchronous loading of options.
+		 */
+		isLoading: bool,
+		/**
+		 * Applies primary color styling to the control when an item is selected. Defaults to true.
+		 */
+		isSelectionHighlighted: bool,
+		/**
+		 * The max height of the fly-out menu.
+		 */
+		maxMenuHeight: oneOfType([number, string]),
+		/**
+		 * Called when the user enters a value to search for; the set of visible Options will be filtered using the value.
+		 * Has the signature `(searchText, {props, event}) => {}` where `searchText` is the new `searchText` or `null`.
+		 */
+		onSearch: func,
+		/**
+		 * Called when an option is selected.
+		 * Has the signature `(optionIndex, {props, event}) => {}` where `optionIndex` is the new `selectedIndex` or `null`.
+		 */
+		onSelect: func,
+		/**
+		 * The function that will be run against each Option's props to determine whether it should be visible or not.
+		 * The default behavior of the function is to match, ignoring case, against any text node descendant of the `Option`.
+		 *
+		 * Has the signature `(searchText, optionProps)`
+		 * If `true`, option will be visible. If `false`, the option will not be visible.
+		 */
+		optionFilter: func,
+		/**
+		 * This function will be run against Options with Rich Content for children to allow for special 'underlining' behavior.
+		 * The default behavior of the function is to just return the rich content as is.
+		 *
+		 * Has the signature `(children, searchText)`
+		 */
+		richChildRenderer: func,
```

Comment:
That's true; I'll remove it.

The original intent of this function was to provide a function to use for managing the underlining of rich child contents. Plus, I just liked the name `richChildRenderer`.